Android/src/org/droidplanner/android/fragments/account/DroneshareLoginFragment.java

```diff
@@ -159,6 +212,50 @@ public void onClick(View v) {
 //        GAUtils.sendEvent(eventBuilder);
 //    }
 
+    private void hideSoftInput() {
+        // Hide the soft keyboard and unfocus any active inputs.
+        final Activity activity = getActivity();
+        View view = activity.getCurrentFocus();
+        if (view != null) {
+            final InputMethodManager inputManager = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
+            if (inputManager != null)
+                inputManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
+        }
+    }
+
+    private Boolean validateField(EditText field) {
+        // Trigger text changed to see if we have any errors:
+        field.setText(field.getText());
+        if (field.getError() != null) {
+            return false; // Invalid, the text field will display an error
+        } else {
+            return true; // Valid
+        }
+    }
+
+    public abstract class TextValidator implements TextWatcher {
```

Comment:
`TextValidator` should be declared as a static inner class since it doesn't use any references from the parent class.